packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,551 @@
+  /// Creates a [FixedExtentScrollController].
+  FixedExtentScrollController({
+    this.initialItem: 0,
+  }) : assert(initialItem != null);
+
+  /// The page to show when first creating the scroll view.
+  ///
+  /// Defaults to 0 and must not be null.
+  final int initialItem;
+
+  /// The currently selected item index that's closest to the center of the viewport.
+  ///
+  /// There are circumstances that this [FixedExtentScrollController] can't know
+  /// the current item. Reading [selectedItem] will throw an [AssertionError] in
+  /// the following cases:
+  ///
+  /// 1. No scroll view is currently using this [FixedExtentScrollController].
+  /// 2. More than one scroll views using the same [FixedExtentScrollController].
+  ///
+  /// The [hasClients] property can be used to check if a scroll view is
+  /// attached prior to accessing [selectedItem].
+  int get selectedItem {
+    assert(
+      positions.isNotEmpty,
+      'FixedExtentScrollController.selectedItem cannot be accessed before a '
+      'scroll view is built with it.',
+    );
+    assert(
+      positions.length == 1,
+      'Multiple scroll views cannot be attached to the same FixedExtentScrollController.',
+    );
+    final _FixedExtentScrollPosition position = this.position;
+    return position.itemIndex;
+  }
+
+  /// Animates the controlled scroll view to the given item index.
+  ///
+  /// The animation lasts for the given duration and follows the given curve.
+  /// The returned [Future] resolves when the animation completes.
+  ///
+  /// The `duration` and `curve` arguments must not be null.
+  Future<Null> animateToItem(int itemIndex, {
+    @required Duration duration,
+    @required Curve curve,
+  }) {
+    final _FixedExtentScrollPosition position = this.position;
+    return position.animateTo(
+      itemIndex * position.itemExtent,
+      duration: duration,
+      curve: curve,
+    );
+  }
+
+  /// Changes which item index is centered in the controlled scroll view.
+  ///
+  /// Jumps the item index position from its current value to the given value,
+  /// without animation, and without checking if the new value is in range.
+  void jumpToItem(int itemIndex) {
+    final _FixedExtentScrollPosition position = this.position;
+    position.jumpTo(itemIndex * position.itemExtent);
+  }
+
+  @override
+  ScrollPosition createScrollPosition(ScrollPhysics physics, ScrollContext context, ScrollPosition oldPosition) {
+    return new _FixedExtentScrollPosition(
+      physics: physics,
+      context: context,
+      initialItem: initialItem,
+      oldPosition: oldPosition,
+    );
+  }
+}
+
+/// Metrics for a [ScrollPosition] to a scroll view with fixed item sizes.
+///
+/// The metrics are available on [ScrollNotification]s generated from a scroll
+/// views such as [ListWheelScrollView]s with a [FixedExtentScrollController] and
+/// exposes the current [itemIndex] and the scroll view's [itemExtent].
+class FixedExtentMetrics extends FixedScrollMetrics {
```

Comment:
It's confusing that the "Fixed" in FixedExtentMetrics and FixedScrollMetrics refer to entirely different things. We should have a paragraph or two here that explains this confusing matter (or we could use a different name, but I don't know what that would be).

Reply:
Ya, this annoyed me too. Luckily this is the only occurrence of `Fixed` that isn't `FixedExtent` in this file.

I can't really not name it `FixedExtent` because it'll be confusing why it's named differently vs `SliverFixedExtentList` and the likes. 

Added a doc comment as suggested.